dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -11,6 +11,8 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
   # TODO: figure out how to avoid skipping CSRF verification for Powerschool
   skip_before_action :verify_authenticity_token, only: :powerschool
 
+  before_action :check_account_linking_lock, on: %i[connect_provider link_accounts]
```

Comment:
I missed this again when I cherry-picked your changes, here. The `on` needs to be `only`. Why does it let you even do that!! 😢